### PR TITLE
Add cheat sheet to the docs

### PR DESCRIPTION
**Pull request by SimeonEhrig**
# About
- The cheat sheet is written with reStructed text
- It can integrate in the website
- Can compiled to pdf a compact format with with `rst2pdf`

# Todo
- [x] integration in Sphinx Doc build
- [x] add build instruction for the cheatsheet pdf to the Sphinx Doc

### CODE REVIEW

**Comment by SimeonEhrig** [comment]
Output of commit 4ab7041a3d530089708751ee23a88bbb9124efb4

[cheatsheet.pdf](https://github.com/alpaka-group/alpaka/files/4852292/cheatsheet.pdf)

**Comment by SimeonEhrig** [comment]
@sbastrakov If you have not already done so, please check the content that I did not make an error in transferring the content from your version.

File: docs/cheatsheet/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# About
+
+The style cheat (cheatsheet.style) was original develop by Roberto Alsina.
```
Comment (sbastrakov) [inline docs/cheatsheet/README.md]: Please replace `original develop` with `originally developed`.
I am also not sure about `style cheat`, perhaps you meant something like `cheat sheet style`?

File: docs/source/usage/cheatsheet.rst [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+Enqueue a memory copy from device to host
+  .. code-block:: c++
+
+     mem::view::copy(queue, bufHost, bufDevice, extent);
+
+.. raw:: pdf
+
+   PageBreak
+
+The Kernel Execution
```
Comment (sbastrakov) [inline docs/source/usage/cheatsheet.rst]: I would remove `The` (and it was not in my version).

File: docs/source/usage/cheatsheet.rst [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+
+Define a kernel as a C++ functor
+  .. code-block:: c++
+
+     struct Kernel {
+        template<typename Acc>
+        ALPAKA_FN_ACC void operator()(Acc const & acc, parameters) const { ... }
+     };
+
+``ALPAKA_FN_ACC`` is required for kernels and functions called inside, acc is mandatory first parameter, its type is the template parameter
```
Comment (sbastrakov) [inline docs/source/usage/cheatsheet.rst]: Perhaps put acc as code?

File: docs/source/usage/cheatsheet.rst [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+  .. code-block:: c++
+
+     block::sync::syncBlockThreads(acc);
+
+Atomic operations
+  .. code-block:: c++
+
+     auto result = atomic::atomicOp<Operation>(acc,
+                                               arguments,
+					       OperationHierarchy);
```
Comment (sbastrakov) [inline docs/source/usage/cheatsheet.rst]: This line seems unaligned to the previous one

File: docs/source/usage/cheatsheet.rst [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+  .. code-block:: c++
+
+     math::sin(acc, argument);
+
+Similar for other math functions.
+
+Generate random numbers
+  .. code-block:: c++
+
+     auto distribution = rand::distribution::createNormalReal<double>(acc));
```
Comment (sbastrakov) [inline docs/source/usage/cheatsheet.rst]: This is also a mistake in my document, should be one `)` in the end, not two.

File: docs/source/usage/cheatsheet.rst [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+					       OperationHierarchy);
+
+  Operation (all in atomic::op):
+     .. code-block:: c++
+
+	namespace atomic::op {
+           Add, Sub, Min, Max, Exch, Inc, Dec, And, Or, Xor, Cas
+	}
+
+  OperationHierarchy (all in hierarchy):
```
Comment (sbastrakov) [inline docs/source/usage/cheatsheet.rst]: I like this change to put stuff in the namespace, but then the brackets (all in ...) is not necessary.

**Review by sbastrakov — CHANGES_REQUESTED** [review]
Almost everything is fine, just a couple of small things in review comments.

**Comment by sbastrakov** [comment]
Also, I think it's fair if I am a co-author of this PR since the original cheat sheet content authorship. Or you can optionally separate your style changes in a separate commit, and add the contents as another one, that is co-authored.

File: docs/source/usage/cheatsheet.rst (hunk 3 above)
Comment (SimeonEhrig) [reply docs/source/usage/cheatsheet.rst]: I highlighted the acc as code

**Comment by SimeonEhrig** [comment]
> Also, I think it's fair if I am a co-author of this PR since the original cheat sheet content authorship. Or you can optionally separate your style changes in a separate commit, and add the contents as another one, that is co-authored.

Yes, it's totally fair, that you are the co-author. I have never tried to add an co-author before, so I hope my solution is fine (see commit message). If you have better solution, please write.

**Comment by sbastrakov** [comment]
@SimeonEhrig I believe there should be a new line before Co-author line. Or somehow this one did not work.

**Comment by sbastrakov** [comment]
Ah, and I use my personal email for github, not work one. Don't want to post it here, but you can see in the log.

**Review by sbastrakov — APPROVED** [review]

**Comment by sbastrakov** [comment]
I contributed to this PR, so would ask for an additional review.

**Review by BenjaminW3 — APPROVED** [review]

**Pull request merged**